src/app/inventory/store/stats.ts

```diff
@@ -216,9 +239,9 @@ function shouldShowStat(
 
   return (
     // Must be on the whitelist
-    statWhiteList.includes(statHash) &&
+    statWhiteListObj[statHash] &&
```

Comment:
I understand most of the other optimizations, but I'm not sure about this one - first, why use an object instead of a Set, and second, are you sure that a straight linear scan of a short list of numbers isn't faster than either?

I threw together a quick benchmark which may be flawed, using a list of numbers of the size we use, searching for something that's near the end of the list. I compared:

* `includes` on the list of numbers
* Your hash object technique
* A Set
* A Map
* A sparse array

`includes` on the list, and the Map, were both 30x faster than the other techniques, which were all about the same. Note also that we're talking about 1.6 billion ops/sec for the list scan vs. 50 million ops/sec for the other techniques - it's unlikely to matter either way and neither makes substantial allocations which would be the real killer.

For the most part I've tried to stick with linear array scans of when I have just numbers and used Sets for big sets of strings or other things that aren't trivially comparable. JS objects in general always stringify keys so you end up paying for a string comparison instead of a straight number comparison. Map allows different key types so it's faster, but in this case we just care about set membership so scanning the list is the best.

Reply:
I re-tested this on Firefox - there, everything's quite a lot slower, and Set (followed closely by Map) is the best. But they're close, and the straight `includes` is still fast compared to the hash object.

Reply:
Interesting - Safari actually doesn't perform too great on the straight search. Set and Map are the clear winners there. So it looks like if raw performance is the only criteria you should use the data structures specifically designed for this task.

But my overall point still stands - whether you're getting 15 million ops/second or 2 billion, these aren't the make or break for performance compared to something heavy like allocation/GC.

Reply:
Your results are interesting, usually a plain object (how we _used_ to implement a map) is faster than a Map. It's possible the performance of a plain object depends on how you create it...

Of course, let's not get too hung up on the microbenchmarking :P